sparkle/src/components/DataTable.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { ReactNode } from "react";
+
+import { Page } from "@sparkle/index";
+import { ClipboardCheckIcon, ClipboardIcon, IconButton } from "@sparkle/index";
+import { useCopyToClipboard } from "@sparkle/lib/utils";
+
+interface DataTableProps {
+  name?: string;
+  columns: string[];
+  rows: (string | number)[][];
+  enableCopy?: boolean;
+  showLimit?: number;
+  downLoadButton?: ReactNode;
+}
+
+export const DataTable = React.forwardRef<HTMLTableElement, DataTableProps>(
+  (
+    {
+      name,
+      columns,
+      rows,
+      enableCopy,
+      showLimit,
+      downLoadButton,
+    }: DataTableProps,
+    ref
+  ) => {
+    const [isCopied, copyToClipboard] = useCopyToClipboard
```

Comment:
Remember hooks consistency! You can't have conditional around hooks. In this case it's fine to always iniatialize.